### DISCUSSION TITLE
Default run-name value when running an Action on a push event.

### DISCUSSION THREAD

**Discussion by g0nP** [community/community#103113] [Actions]
### Select Topic Area

Question

### Body

Hi,

I have an Action that builds my code after a push to the main branch.

```
on:
  push:
    branches: [ 'main' ]
```

I haven't specified any run-name in the workflow file, so in the Actions tab I can see it defaults to the title of the PR I had previously pushed to main.

Does anyone know the exact context/property GitHub uses to get the title of the PR pushed to use it as default value for the run name?

I have found two properties but neither work for me:

- github.event.pull_request.title: this one returns the title of the PR but only when the action triggers for a pull_request event. When the actions triggers for a push event, the property is null.
- github.event.head_commit.message: I squash commits before merging so this property returns a concatenation of the PR title and all its commits, e.g. "The PR Title * Commit 1 message * Commit 2 message * Commit 3 message"

Does anyone know which is the property that would return just "The PR Title" from the concatenation above when I trigger the Action on a push event?

I am looking to form a custom run-name that includes "The PR Title" plus some other information.

Thank you.

**Comment by onkarbhujbal** [comment]
I see issue is resolved. Can I know what is the exact context/property GitHub uses to get the title of the PR pushed to use it as the default value for the run name?
